### PR TITLE
APF: use snake_case in metric labels

### LINKED ISSUE #96233: APF metrics: use snake_case for metric labels instead of camelCase
**What would you like to be added**: API priority and fairness metrics have labels such as "priorityLevel" and "flowSchema". These should be "priority_level" and "flow_schema" respectively. These metrics have been added to an exception list:

https://github.com/kubernetes/kubernetes/blob/396b90f06ca29c754a81eb6c03cff50b7d9bf265/staging/src/k8s.io/component-base/metrics/testutil/promlint.go#L36-L39

**Why is this needed**: snake_case is consistent with other metrics.

/sig api-machinery
/kind cleanup
/cc @MikeSpreitzer @ahg-g 

Do we want to do this before 1.20 beta?

### PR DESCRIPTION
**Pull request by adtac**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**: APF metrics should use snake_case for label names, not camelCase, as per convention and linter requirements. The feature is currently alpha (moving to beta in 1.20), so it's safe to do this.

**Which issue(s) this PR fixes**:

Fixes https://github.com/kubernetes/kubernetes/issues/96233

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
API priority and fairness metrics use snake_case in label names
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

/sig api-machinery
/cc @ahg-g @MikeSpreitzer @deads2k

### CODE REVIEW

**Review by MikeSpreitzer — APPROVED** [review]
/LGTM
Thanks!

**Comment by MikeSpreitzer** [comment]
We should keep the doc consistent; see https://github.com/kubernetes/website/issues/24895

File: staging/src/k8s.io/apiserver/pkg/util/flowcontrol/metrics/metrics.go [hunk 1]
```diff
@@ -34,8 +34,8 @@ const (
 
 const (
 	requestKind   = "request_kind"
-	priorityLevel = "priorityLevel"
-	flowSchema    = "flowSchema"
+	priorityLevel = "priority_level"
+	flowSchema    = "flow_schema"
```
Comment (sttts) [inline staging/src/k8s.io/apiserver/pkg/util/flowcontrol/metrics/metrics.go:38]: did we forget to set the StabilityLevel everywhere in the metrics below?

Comment (adtac) [reply staging/src/k8s.io/apiserver/pkg/util/flowcontrol/metrics/metrics.go:38]: Done. Set all metrics to ALPHA.

**Comment by adtac** [comment]
docs PR: https://github.com/kubernetes/website/pull/24905

**Review by MikeSpreitzer — APPROVED** [review]
/LGTM

**Pull request merged**